wire-compiler/src/main/java/com/squareup/wire/OutputArtifact.java

```diff
@@ -0,0 +1,44 @@
+package com.squareup.wire;
+
+import java.io.File;
+
+/**
+ * A container class that represents an artifact output from the compiler.
+ */
+public class OutputArtifact {
```

Comment:
Derived methods work with AutoValue. The dependency is compile-time only. There is no runtime component.

I'm also not convinced on point 1, just because we aren't doing something doesn't mean we shouldn't include correct implementations of those methods. Ultimately this is a value type so not being able to put it in a `Set` or use `equals` comparisons seems incorrect.